jscomp/ml/dict_type_helpers.ml

```diff
@@ -0,0 +1,48 @@
+(*
+  An overview of the implementation of dicts in ReScript:
+  ### What is a dict?
+  Dicts are effectively an object with unknown fields, but a single known type of the values it holds.
+
+  ### How are they implemented?
+  Dicts in ReScript are implemented as predefined record type, with a single (magic) field that holds 
```

Comment:
perhaps somewhere: say explicitly that it represents every possible key in the dict